flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/source/split/BigQuerySourceSplitState.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.flink.bigquery.source.split;
+
+import org.apache.flink.annotation.Internal;
+
+import java.util.Objects;
+
+/** BigQuery source split state for {@link BigQuerySourceSplit}. */
+@Internal
+public class BigQuerySourceSplitState {
+    private final BigQuerySourceSplit split;
+    private Integer offset;
+
+    public BigQuerySourceSplitState(BigQuerySourceSplit split) {
+        this.split = split;
+        offset = split.getOffset();
+    }
+
+    public BigQuerySourceSplit toBigQuerySourceSplit() {
+        return new BigQuerySourceSplit(split.getStreamName(), offset);
+    }
+
+    public void updateOffset() {
+        offset++;
+    }
+
+    @Override
+    public String toString() {
+        return "BigQuerySourceSplitState{" + "split=" + split + ", offset=" + offset + '}';
+    }
+
+    @Override
+    public int hashCode() {
+        int hash = 7;
+        hash = 53 * hash + Objects.hashCode(this.split);
+        hash = 53 * hash + Objects.hashCode(this.offset);
+        return hash;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        final BigQuerySourceSplitState other = (BigQuerySourceSplitState) obj;
+        if (!Objects.equals(this.split, other.split)) {
+            return false;
+        }
+        return Objects.equals(this.offset, other.offset);
```

Comment:
This will return false if two integer objects, even with the same value, right? I guess what we want here is just to check whether the two offset values are same.

Reply:
Not sure if I understood this comment, but `Objects.equals` methods does check on the values boxed in the Integer objects.

```
$ jshell
|  Welcome to JShell -- Version 11.0.15
|  For an introduction type: /help intro

jshell> Integer a = 10
a ==> 10

jshell> Integer b = 10
b ==> 10

jshell> import java.util.Objects;

jshell> Objects.equals(a, b)
$4 ==> true
```

JIC, adding tests to corroborate that.